akka-docs/src/main/paradox/stream/operators/Sink/foreachAsync.md

```diff
@@ -0,0 +1,38 @@
+# foreachAsync
+
+Invoke a given procedure asynchronously for each element received.
+
+@ref[Sink operators](../index.md#sink-operators)
+
+@@@div { .group-scala }
+
+## Signature
+
+@@signature [Sink.scala]($akka$/akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala) { #foreachAsync }
+
+@@@
+
+## Description
+
+Invoke a given procedure asynchronously for each element received. Note that if shared state is mutated from the procedure that must be done in a thread-safe way.
+
+The sink materializes into a  @scala[`Future[Done]`] @java[`CompletionStage<Done>`] which completes when the
+stream completes, or fails if the stream fails.
+
+## Example
+
+Scala
+:   @@snip [SinkRecipeDocSpec.scala]($code$/scala/docs/stream/SinkRecipeDocSpec.scala) { #forseachAsync-processing }
+
+Java
+:   @@snip [SinkRecipeDocTest.java]($code$/java/jdocs/stream/SinkRecipeDocTest.java) { #forseachAsync-processing }
+
+@@@div { .callout }
+
+**cancels** never
```

Comment:
not even when the procedure throws or if the Future is completed with exception?